packages/babel-traverse/src/path/family.js

```diff
@@ -17,6 +17,71 @@ function addCompletionRecords(path, paths) {
   return paths;
 }
 
+function completionRecordForSwitch(cases, paths) {
+  for (const switchcase of cases) {
+    const consequentLength = switchcase.get("consequent").length;
+    if (consequentLength === 0) {
+      continue;
+    }
+    const isDefaultCase = switchcase.get("test").type === null;
+    if (isDefaultCase) {
+      paths = addCompletionRecords(
+        switchcase.get("consequent")[consequentLength - 1],
+        paths,
+      );
+    } else {
+      const lastCaseStatement = switchcase.get("consequent")[
+        consequentLength - 1
+      ];
+      const hasBreakStatement =
+        lastCaseStatement && lastCaseStatement.isBreakStatement();
+
+      if (hasBreakStatement) {
+        if (consequentLength === 1) {
+          lastCaseStatement.replaceWith(switchcase.scope.buildUndefinedNode());
+          paths = addCompletionRecords(switchcase.get("consequent")[0], paths);
+        }
+        if (consequentLength > 1) {
+          paths = addCompletionRecords(
+            switchcase.get("consequent")[consequentLength - 2],
+            paths,
+          );
+          lastCaseStatement.remove();
+        }
+      }
+
+      if (!hasBreakStatement) {
+        if (consequentLength >= 1) {
+          paths = addCompletionRecords(
+            switchcase.get("consequent")[consequentLength - 1],
+            paths,
+          );
+        }
+      }
+
+      if (hasBreakStatement && consequentLength === 1) {
+        lastCaseStatement.replaceWith(switchcase.scope.buildUndefinedNode());
+        paths = addCompletionRecords(switchcase.get("consequent")[0], paths);
+      }
+
+      if (hasBreakStatement && consequentLength > 1) {
+        paths = addCompletionRecords(
+          switchcase.get("consequent")[consequentLength - 1],
+          paths,
+        );
+      }
+
+      if (!hasBreakStatement && consequentLength) {
```

Comment:
Only if the first statement of the following `case` is break.

e.g.
```js
let a = do {
  switch(0) {
    case 0:
      "foo";
    case 1:
      "bar";
      break;
  }
};
```

Should be "bar".

Reply:
Yes right.